pkg/controllers/leaderworkerset_controller.go

```diff
@@ -582,6 +663,10 @@ func constructLeaderStatefulSetApplyConfiguration(lws *leaderworkerset.LeaderWor
 			podAnnotations[leaderworkerset.SubGroupExclusiveKeyAnnotationKey] = lws.Annotations[leaderworkerset.SubGroupExclusiveKeyAnnotationKey]
 		}
 	}
+
+	if lws.Spec.NetworkConfig != nil && lws.Spec.NetworkConfig.SubdomainPolicy == leaderworkerset.SubdomainUniquePerReplica {
```

Comment:
you will also not need this annotation if we do deletions via ownerReference

Reply:
The annotation is there to determine what domain to inject on the leader. Still need it even setting the worker stateful set as the owner.